lib/ansible/modules/packaging/os/redhat_subscription.py

```diff
@@ -515,7 +515,10 @@ def subscribe_by_pool_ids(self, pool_ids):
 
         for pool_id, quantity in sorted(pool_ids.items()):
             if pool_id in available_pool_ids:
-                args = [SUBMAN_CMD, 'attach', '--pool', pool_id, '--quantity', quantity]
+                if quantity is not None:
+                    args = [SUBMAN_CMD, 'attach', '--pool', pool_id, '--quantity', str(quantity)]
+                else:
+                    args = [SUBMAN_CMD, 'attach', '--pool', pool_id]
```

Comment:
This can be further simplified by only adding the quanity flags when needed.
```python
                args = [SUBMAN_CMD, 'attach', '--pool', pool_id]
                if quantity is not None:
                    args.extend(['--quantity', to_native(quantity)])
```

Avoid using `str()` since it returns a different type on Python 2 or Python 3. Use `to_native()` instead to ensure the correct string type is used for the current Python version.

Reply:
Thanks for review. It should be simplified and fixed ATM.